src/components/data/PowerBIReport/index.tsx

```diff
@@ -0,0 +1,274 @@
+import * as React from 'react';
+import * as pbi from 'powerbi-client';
+import { IError } from 'powerbi-models';
+import { Spinner, ErrorMessage } from '@equinor/fusion-components';
+import {
+    useTelemetryLogger,
+    FusionApiHttpErrorResponse,
+    useApiClients,
+    useCurrentApp,
+} from '@equinor/fusion';
+import { ReportType } from './models/reportTypes';
+import { ICustomEvent } from 'service';
+import FusionError from './models/FusionError';
+import FusionErrorMessage from './components/FusionErrorMessage';
+import { Report, EmbedInfo, AccessToken } from '@equinor/fusion/lib/http/apiClients/models/report/';
+
+import * as styles from './styles.less';
+
+type PowerBIProps = {
+    reportId: string;
+    filters?: pbi.models.ReportLevelFilters[] | null;
+};
+
+const getReportOrDashboardId = (embedConfig: any, type: ReportType) => {
+    switch (type) {
+        case 'report':
+            return embedConfig.reportId;
+        case 'dashboard':
+            return embedConfig.dashboardId;
+    }
+};
+
+const getTokenType = (embedConfig: any) => {
+    switch (embedConfig.tokenType) {
+        case 'AAD':
+            return pbi.models.TokenType.Aad;
+        case 'Embed':
+            return pbi.models.TokenType.Embed;
+    }
+};
+
+const powerbi = new pbi.service.Service(
+    pbi.factories.hpmFactory,
+    pbi.factories.wpmpFactory,
+    pbi.factories.routerFactory
+);
+
+const utcNow = () => {
+    const date = new Date();
+
+    const utc = new Date(
+        date.getUTCFullYear(),
+        date.getUTCMonth(),
+        date.getUTCDate(),
+        date.getUTCHours(),
+        date.getUTCMinutes(),
+        date.getUTCSeconds(),
+        date.getUTCMilliseconds()
+    );
+
+    return new Date(utc);
+};
+
+let timeout: any;
+
+const PowerBIReport: React.FC<PowerBIProps> = ({ reportId, filters }) => {
+    const reportContext = useApiClients().report;
+
+    const [isLoading, setIsLoading] = React.useState<boolean>(true);
+    const [isFetching, setIsFetching] = React.useState<boolean>(true);
+    const [powerBIError, setPowerBIError] = React.useState<ICustomEvent<IError>>();
+    const [fusionError, setFusionError] = React.useState<FusionError>();
+
+    const [report, setReport] = React.useState<Report>();
+    const [embedInfo, setEmbedInfo] = React.useState<EmbedInfo>();
+    const [accessToken, setAccessToken] = React.useState<AccessToken>();
+    const telemetryLogger = useTelemetryLogger();
+
+    const embedRef = React.useRef<HTMLDivElement>(null);
+    const embeddedRef = React.useRef<pbi.Embed | null>(null);
+    const timeLoadStart = new Date();
```

Comment:
This should probably be stored with useRef or useState. Just setting it to a variable will reset it each render